packages/ui-components/src/components/Chart/GaugeChart.js

```diff
@@ -0,0 +1,162 @@
+/*
+ * Tupaia
+ * Copyright (c) 2017 - 2022 Beyond Essential Systems Pty Ltd
+ */
+
+/**
+ * GaugeChart
+ *
+ * Renders a recharts GaugeChart from data provided by viewContent object
+ * @prop {object} viewContent An object with the following structure
+   {
+    "type":"chart",
+    "chartType":"gauge",
+    "name":"% Stock on Hand",
+    "valueType": "percentage",
+    "color": "#111111",
+    "data":[{ value:0.485 }]
+  }
+ */
+
+import React, { useCallback, useMemo } from 'react';
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+import { formatDataValueByType } from '@tupaia/utils';
+import {
+  PieChart as BasePieChart,
+  Pie,
+  Cell,
+  ResponsiveContainer,
+  Label,
+  Text as RechartText,
+} from 'recharts';
+import { VIEW_CONTENT_SHAPE, BLUE, TRANS_BLACK, WHITE } from './constants';
+import { isMobile } from './utils';
+
+const Text = styled(RechartText)`
+  font-size: ${p => p.$fontSize};
+  font-weight: bold;
+  fill: ${({ theme, $isExporting }) => {
+    return theme.palette.type === 'light' || $isExporting ? TRANS_BLACK : WHITE;
+  }};
+`;
+
+const getHeight = (isExporting, isEnlarged) => {
+  if (isExporting) {
+    return 420;
+  }
+  return isEnlarged && isMobile() ? 320 : undefined;
+};
+
+export const GaugeChart = props => {
+  const { viewContent, isExporting, isEnlarged, onItemClick } = props;
+  const { data, color = BLUE, ...restOfConfigs } = viewContent;
+
+  const generateElements = () => {
+    const denominator = 0.05;
+    const elements = [...data];
+    const cellComponents = [<Cell fill={color} />];
+    const numOfElements = Math.floor((1 - data[0].value) / denominator);
+
+    for (let i = 0; i < numOfElements; i++) {
+      elements.push({ value: denominator });
+      cellComponents.push(<Cell key={`cell-${i}`} fill="#eaeaea" />);
+    }
+
+    return { elements, cellComponents };
+  };
+
+  const { elements, cellComponents } = useMemo(() => generateElements(), [data]);
+  const height = useMemo(() => getHeight(isExporting, isEnlarged), [isExporting, isEnlarged]);
+
+  const responsiveStyle = !isEnlarged && !isMobile() && !isExporting ? 1 : 1.5;
```

Comment:
Positive conditions are generally easier to understand than negative
```suggestion
  const responsiveStyle = isEnlarged || isMobile() || isExporting ? 1.5 : 1;
```

Reply:
That makes sense!